File: JS/02.Basics/02.3-ConditionalStatementsExercise/08.LunchBreak.js
function lunchBreak(arr) {
    let seriesName = arr[0];
    let episodeDuration = parseInt(arr[1]);
    let breakDuration = parseInt(arr[2]);

    let lunchTime = breakDuration * 1/8;
    let restTime = breakDuration * 1/4;
    let remainingTime = breakDuration - lunchTime - restTime;

    if (remainingTime >= episodeDuration) {
        let leftFreeTime = Math.ceil(remainingTime - episodeDuration);
        console.log(`You have enough time to watch ${seriesName} and left with ${leftFreeTime} minutes free time.`);
    } else {
        let neededTime = Math.ceil(episodeDuration - remainingTime);
        console.log(`You don't have enough time to watch ${seriesName}, you need ${neededTime} more minutes.`);
    }
}

lunchBreak(["Game of Thrones", "60", "96"]); 
lunchBreak(["Teen Wolf", "48", "60"]); 

File: C++/01-Basics/02.3-ConditionalStatementsExercise/08.LunchBreak/08.LunchBreak.cpp
#include <iostream>
#include <cmath>
#include <string>

using namespace std;

int main() {
    string seriesName;
    getline(cin, seriesName);

    int episodeDuration, breakDuration;
    cin >> episodeDuration >> breakDuration;

    cin.ignore(numeric_limits<streamsize>::max(), '\n');

    double lunchTime = breakDuration * 1.0 / 8;
    double restTime = breakDuration * 1.0 / 4;
    double remainingTime = breakDuration - lunchTime - restTime;

    if (remainingTime >= episodeDuration) {
        cout << "You have enough time to watch " << seriesName
            << " and left with " << static_cast<int>(round(remainingTime - episodeDuration))
            << " minutes free time." << endl;
    }
    else {
        cout << "You don't have enough time to watch " << seriesName
            << ", you need " << static_cast<int>(round(episodeDuration - remainingTime))
            << " more minutes." << endl;
    }

    return 0;
}
